static/app/components/duration/duration.tsx

```diff
@@ -0,0 +1,37 @@
+import type {HTMLAttributes} from 'react';
+import {css} from '@emotion/react';
+
+import formatDuration from 'sentry/utils/duration/formatDuration';
+import type {Duration as TDuration, Unit} from 'sentry/utils/duration/types';
+
+interface Props extends HTMLAttributes<HTMLTimeElement> {
+  /**
+   * The Duration that you want to render
+   */
+  duration: TDuration;
+
+  /**
+   * How granular to render the value. You can pass in something that has `ms`
+   * precision but only show the total number of seconds (ala Math.floor)
+   */
+  precision: Unit;
+}
+
+export default function Duration({duration, precision, ...props}: Props) {
```

Comment:
props here has a massive custom type
![image](https://github.com/user-attachments/assets/aacd7ba7-30e0-429d-823a-0b009c7e998b)

might be another way to avoid this

Reply:
or Pick just the properties you need

Reply:
I'm not super worried about this, we already mostly extend HTML types. It's only when we start having large intersections that it becomes a big problem

Reply:
this does get type checked when merged back into the time element and will take quite a bit of time